### PR TITLE
WFLY-5520 Manually migrated EAP 6.x profiles fail to boot due to 'default-clustered-sfsb-cache' used in runtime

### PR DESCRIPTION
**Pull request by rhusar**
Need to discuss with Brian first.

The idea here is that in EAP7 all SFSB are clustered by default. So in the EAP7 default config, cache-ref is no longer simple, but clustered and thus there is no longer clustered-cache-ref. However, passivation-disabled-cache-ref is introduced which is used for `@Stateful(passivationCapable = false)` as per spec, those can never be clustered as per spec. So in fact, these could map to each other: clustered caches and non-clustered caches:
- EAP6 cache-ref => EAP7 passivation-disabled-cache-ref 
- EAP6 clustered-cache-ref => EAP7 cache-ref

We can do the differentiation according to the xml schema version, this PR attempts to do so.

Some more ideas on the Jira:
https://issues.jboss.org/browse/WFLY-5520

### CODE REVIEW

**Comment by rhusar** [comment]
retest this please

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemRootResourceDefinition.java [hunk 1]
```diff
@@ -101,19 +101,13 @@
     static final SimpleAttributeDefinition DEFAULT_SFSB_CACHE =
             new SimpleAttributeDefinitionBuilder(EJB3SubsystemModel.DEFAULT_SFSB_CACHE, ModelType.STRING, true)
                     .setAllowExpression(true)
+                    .addAlternatives(EJB3SubsystemModel.DEFAULT_CLUSTERED_SFSB_CACHE)
```
Comment (rhusar) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemRootResourceDefinition.java]: This is wrong.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemRootResourceDefinition.java [hunk 2]
```diff
@@ -101,19 +101,13 @@
...
                     .build();
     static final SimpleAttributeDefinition DEFAULT_SFSB_PASSIVATION_DISABLED_CACHE =
             new SimpleAttributeDefinitionBuilder(EJB3SubsystemModel.DEFAULT_SFSB_PASSIVATION_DISABLED_CACHE, ModelType.STRING, true)
                     .setXmlName(EJB3SubsystemXMLAttribute.PASSIVATION_DISABLED_CACHE_REF.getLocalName())
                     .setAllowExpression(true)
                     .build();
-    static final SimpleAttributeDefinition DEFAULT_CLUSTERED_SFSB_CACHE =
```
Comment (rhusar) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemRootResourceDefinition.java]: This needs to be restored as an alias writing to DEFAULT_SFSB_CACHE.

File: ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Subsystem20Parser.java [hunk 3]
```diff
@@ -132,6 +134,14 @@ protected void parseStatefulBean(final XMLExtendedStreamReader reader, final Lis
             }
             // found the mandatory attribute
             missingRequiredAttributes.remove(attribute);
+
+            // differentiate between non-HA and HA profile
+            if (clusteredCacheRef == null) {
+                EJB3SubsystemRootResourceDefinition.DEFAULT_SFSB_CACHE.parseAndSetParameter(cacheRef, ejb3SubsystemAddOperation, reader);
```
Comment (rhusar) [inline ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3Subsystem20Parser.java]: Suggestion from Brian: we can throw in a friendly INFO that we are doing some sort of mapping here.

**Comment by bstansberry** [comment]
#8610 got merged.

**Pull request closed without merging**